Loosen `these` bounds to 0.7.7?

Now that reflex-0.5 is on Hackage, nixpkgs master has it in hackage-packages.nix. Unfortunately, the version of `these` in nixpkgs is 0.7.6.

@luigy Recently tightened the upper bound for `these` from `< 0.8` to `< 0.7.6`. I'm having trouble driving the reflex build systems to actually test the relaxation. Is 0.7.6 safe for reflex, and if so, can someone please push a metadata rev to hackage?

It looks as though reflex 0.5 breaks with that version of `these`:

```
src/Reflex/Patch/MapWithMove.hs:173:51: error:
    Ambiguous occurrence ‘swap’
    It could refer to either ‘Data.These.swap’,
                             imported from ‘Data.These’ at src/Reflex/Patch/MapWithMove.hs:25:1-17
                          or ‘Data.Tuple.swap’,
                             imported from ‘Data.Tuple’ at src/Reflex/Patch/MapWithMove.hs:26:1-17
    |
173 |   where oldByValue = Map.fromListWith Set.union $ swap . first Set.singleton <$> Map.toList oldByIndex
    |                                                   ^^^^
```

so a new hackage release with a code change would be necessary to widen that bound.

--- a/reflex.cabal
+++ b/reflex.cabal
@@ -60,7 +60,7 @@ library
     semigroups >= 0.16 && < 0.19,
     stm >= 2.4 && < 2.6,
     syb >= 0.5 && < 0.8,
-    these >= 0.4 && < 0.7.6,
+    these >= 0.4 && < 0.7.7,
     time >= 1.4 && < 1.9,
     transformers >= 0.2,
     transformers-compat >= 0.3,

--- a/src/Reflex/Patch/MapWithMove.hs
+++ b/src/Reflex/Patch/MapWithMove.hs
@@ -22,7 +22,7 @@ import qualified Data.Map as Map
 import Data.Maybe
 import Data.Semigroup (Semigroup (..), (<>))
 import qualified Data.Set as Set
-import Data.These
+import Data.These (These(..))
 import Data.Tuple
 
 -- | Patch a DMap with additions, deletions, and moves.  Invariant: If key @k1@

